hosted-cluster-config-operator/controllers/resources/resources.go

```diff
@@ -104,8 +157,179 @@ func (r *reconciler) reconcile(ctx context.Context) error {
 		kubeletServingCAConfigMap.Data = map[string]string{"ca-bundle.crt": r.clusterSignerCA}
 		return nil
 	}); err != nil {
-		return fmt.Errorf("failed to reconcile the %s ConfigMap: %w", crclient.ObjectKeyFromObject(kubeletServingCAConfigMap), err)
+		errs = append(errs, fmt.Errorf("failed to reconcile the %s ConfigMap: %w", crclient.ObjectKeyFromObject(kubeletServingCAConfigMap), err))
 	}
 
-	return nil
+	log.Info("reconciling kube apiserver service monitor")
+	kasServiceMonitor := manifests.KubeAPIServerServiceMonitor()
+	if _, err := r.CreateOrUpdate(ctx, r.client, kasServiceMonitor, func() error {
+		return monitoring.ReconcileKubeAPIServerServiceMonitor(kasServiceMonitor)
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile the kube apiserver service monitor: %w", err))
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+func (r *reconciler) reconcileCRDs(ctx context.Context) error {
+	var errs []error
+
+	requestCount := manifests.RequestCountCRD()
+	if _, err := r.CreateOrUpdate(ctx, r.client, requestCount, func() error {
+		return crd.ReconcileRequestCountCRD(requestCount)
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile request count crd: %w", err))
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+func (r *reconciler) reconcileConfig(ctx context.Context, hcp *hyperv1.HostedControlPlane, globalConfig globalconfig.GlobalConfig) error {
+	var errs []error
+
+	apiServerAddress := hcp.Status.ControlPlaneEndpoint.Host
+
+	if len(apiServerAddress) == 0 {
+		return fmt.Errorf("hosted control plane does not have an APIServer endpoint address")
+	}
+
+	// Infrastructure is first reconciled for its spec
+	infra := globalconfig.InfrastructureConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, infra, func() error {
+		globalconfig.ReconcileInfrastructure(infra, hcp)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile infrastructure config spec: %w", err))
+	}
+	// It is reconciled a second time to update its status
+	globalconfig.ReconcileInfrastructure(infra, hcp)
```

Comment:
Maybe create a deepcopy first and check if the reconciled status differs from the one that is already in the api and if so, skip the status update?

I was also at some point thinking about adding code to the CreateOrUpdate provider to make it deal with status updates, by giving it a list of resources that have a status subresource and if it reconciles something that is in that list, make it handle the status.